samples/bot-configuration-app/csharp/Bot configuration/Bots/TeamsBot.cs

```diff
@@ -0,0 +1,286 @@
+                Content = JsonConvert.DeserializeObject(adaptiveCardJson)
+            };
+
+            var reply = MessageFactory.Attachment(attachment);
+            await turnContext.SendActivityAsync(reply, cancellationToken);
+        }
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Text != null)
+            {
+                string text = turnContext.Activity.Text.ToLower().Trim();
+                if (text == "chosen flow" || text == "<at>typeahead search adaptive card</at> chosen flow")
+                {
+                    await turnContext.SendActivityAsync($"Bot configured for {_chosenFlow} flow", cancellationToken: cancellationToken);
+                }
+            }
+            else if (turnContext.Activity.Value != null)
+            {
+                var choiceselect = turnContext.Activity.Value;
+                await turnContext.SendActivityAsync($"Selected option is: {choiceselect}", cancellationToken: cancellationToken);
+            }
+        }
+
+        protected override Task<ConfigResponseBase> OnTeamsConfigFetchAsync(ITurnContext<IInvokeActivity> turnContext, JObject configData, CancellationToken cancellationToken)
+        {
+            /*
+            Option 1: You can add a "config/auth" response as below code
+            Note: The URL in value must be linked to a valid auth URL which can be opened in a browser. This code is only representative and not a working example.
+            */
+            /*ConfigResponseBase response = new ConfigResponse<BotConfigAuth>
+            {
+                Config = new BotConfigAuth
+                {
+                    SuggestedActions = new SuggestedActions
+                    {
+                        Actions = new List<CardAction>
+                {
+                    new CardAction
+                    {
+                        Type = "openUrl",
+                        Title = "Sign in to this app",
+                        Value = "https://example.com/auth"
+                    }
+                }
+                    },
+                    Type = "auth"
+                }
+            };*/
+
+            ConfigResponseBase response = adaptiveCardForContinue();
+
+            return Task.FromResult(response);
+        }
+
+        protected override Task<ConfigResponseBase> OnTeamsConfigSubmitAsync(ITurnContext<IInvokeActivity> turnContext, JObject configData, CancellationToken cancellationToken)
+        {
+            JObject choice = JsonConvert.DeserializeObject<JObject>(turnContext.Activity.Value.ToString());
+            try
+            {
+                _chosenFlow = choice["data"]["choiceselect"].ToString();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+
+            if (_chosenFlow == "continue")
```

Comment:
should we mention it like option 1 and option 2 as well so it aligns with the config/fetch section.

Reply:
No, we need not to. It will align with config/fetch section as same as in nodeJS.

Fixed